pkg/build/controller/config_controller.go

```diff
@@ -33,11 +39,66 @@ func IsFatal(err error) bool {
 
 type BuildConfigController struct {
 	BuildConfigInstantiator buildclient.BuildConfigInstantiator
+
+	KubeClient kclient.Interface
+	Client     osclient.Interface
+
+	JenkinsConfig serverapi.JenkinsPipelineConfig
+
+	// recorder is used to record events.
+	Recorder record.EventRecorder
 }
 
 func (c *BuildConfigController) HandleBuildConfig(bc *buildapi.BuildConfig) error {
 	glog.V(4).Infof("Handling BuildConfig %s/%s", bc.Namespace, bc.Name)
 
+	if strategy := bc.Spec.Strategy.JenkinsPipelineStrategy; strategy != nil {
+		svcName := c.JenkinsConfig.ServiceName
+		if len(svcName) == 0 {
+			return fmt.Errorf("the Jenkins Pipeline ServiceName must be set in master configuration")
+		}
+
+		glog.V(4).Infof("Detected Jenkins pipeline strategy in %s/%s build configuration", bc.Namespace, bc.Name)
+		if _, err := c.KubeClient.Services(bc.Namespace).Get(svcName); err == nil {
```

Comment:
The reconciliation aspect depends on more pieces being in play - we have to
solve that stuff, but it doesn't block the iteration here.  "Delete yo
stuff" usually works.

On Apr 29, 2016, at 10:46 PM, Jordan Liggitt <notifications@github.com>
wrote:

In pkg/build/controller/config_controller.go
<https://github.com/openshift/origin/pull/8693#discussion_r61663718>:

>  }
>
>  func (c *BuildConfigController) HandleBuildConfig(bc *buildapi.BuildConfig) error {
>  	glog.V(4).Infof("Handling BuildConfig %s/%s", bc.Namespace, bc.Name)
>
> +	if strategy := bc.Spec.Strategy.JenkinsPipelineStrategy; strategy != nil {
> +		svcName := c.JenkinsConfig.ServiceName
> +		if len(svcName) == 0 {
> +			return fmt.Errorf("the Jenkins Pipeline ServiceName must be set in master configuration")
> +		}
> +
> +		glog.V(4).Infof("Detected Jenkins pipeline strategy in %s/%s build configuration", bc.Namespace, bc.Name)
> +		if _, err := c.KubeClient.Services(bc.Namespace).Get(svcName); err == nil {

only triggering off the service means that other parts of the template
(backing pod/rc, etc) could be deleted, or the service could be messed up,
and the system won't recover

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/8693/files/5621fa047ef7c1f187d1faa658a125a9586c5357#r61663718>